src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -407,6 +409,7 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		}
 		this._onCommandStarted.fire({ marker: this._currentCommand.commandStartMarker } as ITerminalCommand);
 		this._logService.debug('CommandDetectionCapability#_handleCommandStartWindows', this._currentCommand.commandStartX, this._currentCommand.commandStartMarker?.line);
+		this._commandStartedWindowsBarrier.open();
```

Comment:
This isn't quite right, we want the task that runs when the polling has finished to be after `this._commandStartedWindowsBarrier.wait()`. Right now it will automatically open after 200ms which isn't what we're after as then executed code will run before started code